tests/unit/backend/chalice/api_server/test_revisions.py

```diff
@@ -232,57 +231,82 @@ def test__delete_published_dataset_during_revision(self):
 
-        # Published assets are not deleted
-        self.assertPublishedCollectionOK(expected_body, pub_s3_objects)
+    def publish_collection(self, collection_id: str):
+        self.session.expire_all()
+        path = f"/dp/v1/collections/{collection_id}/publish"
+        response = self.app.post(path, self.headers)
+        response.raise_for_status()
+        self.assertEqual(202, response.status_code)
+        self.assertDictEqual({"collection_uuid": collection_id, "visibility": "PUBLIC"}, json.loads(response.body))
+        self.addCleanup(self.delete_collection, collection_id, "PUBLIC")
 
-    def assertPublishedCollectionOK(self, expected_body, s3_objects):
-        with self.subTest("published artifacts and deployment_directories ok"):
-            for bucket, file_name in s3_objects:
-                self.assertS3FileExists(bucket, file_name)
-        with self.subTest("publish collection ok"):
-            resp = self.app.get(self.test_url_collection_public)
-            resp.raise_for_status()
-            actual_body = json.loads(resp.body)
-            for key in expected_body.keys():
-                if key in actual_body.keys():
-                    self.assertEqual(expected_body[key], actual_body[key])
+        # cannot call twice
+        response = self.app.post(path, self.headers)
+        self.assertEqual(403, response.status_code)
 
-    def refresh_datasets(self):
-        for dataset in self.rev_collection.datasets:
-            Dataset(dataset).delete()
-        for dataset in self.pub_collection.datasets:
-            self.generate_dataset_with_s3_resources(
-                self.session,
-                collection_visibility="PRIVATE",
-                collection_id=self.rev_collection.id,
-                original_id=dataset.id,
-            )
+        # check if the collection is listed
+        path = "/dp/v1/collections"
+        headers = {"host": "localhost", "Content-Type": "application/json"}
+        response = self.app.get(path, headers)
+        response.raise_for_status()
+        ids = [col["id"] for col in json.loads(response.body)["collections"]]
+        self.assertIn(collection_id, ids)
 
-    def get_s3_objects_from_collections(self) -> typing.Tuple[typing.List, typing.List]:
-        """
-        :return: a list of s3 objects in the published collection, and a list of s3 objects the revision collection.
-        """
-        rev_s3_objects = []
-        pub_s3_objects = []
+        # check get collection_uuid
+        path = f"/dp/v1/collections/{collection_id}"
+        headers = {"host": "localhost", "Content-Type": "application/json"}
+        response = self.app.get(path, headers)
+        response.raise_for_status()
+        actual = json.loads(response.body)
+        self.assertEqual("PUBLIC", actual["visibility"])
+        self.assertEqual(collection_id, actual["id"])
+        return actual
+
+    def verify_datasets(self, actual_body, expected_dataset_ids):
```

Comment:
Would it be worth putting this function in `BaseRevisionTest`

Reply:
We can move it once another test needs it.